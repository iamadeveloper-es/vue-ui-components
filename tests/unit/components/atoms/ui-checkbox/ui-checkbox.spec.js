import { createLocalVue, shallowMount } from "@vue/test-utils";
import UiCheckbox from '@/components/atoms/ui-checkbox/index.vue';

describe('Component UiBadge', () => {
    const localVue = createLocalVue();

    const component = shallowMount(UiCheckbox, {
        localVue
    });

    beforeEach(() => {
        jest.clearAllMocks();
    });

    describe('... Definition', () => {

        test('The name of the component is ui-checkbox', () => {
            expect(UiCheckbox.name).toEqual('ui-checkbox');
        });

        test('Match snapshot', () => {
            expect(component.element).toMatchSnapshot();
        });
    });
});