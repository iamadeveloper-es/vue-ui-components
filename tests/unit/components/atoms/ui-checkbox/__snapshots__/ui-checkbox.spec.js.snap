// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UiBadge ... Definition Match snapshot 1`] = `
<div
  class="ui-checkbox"
>
  <div
    class="ui-checkbox__wrapper"
  >
    <span
      class="ui-checkbox__icon"
      ria-hidden="true"
    >
      <svg
        aria-hidden="true"
        class="v-icon__svg"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19,3H5C3.89,3 3,3.89 3,5V19C3,20.1 3.9,21 5,21H19C20.1,21 21,20.1 21,19V5C21,3.89 20.1,3 19,3M19,5V19H5V5H19Z"
        />
      </svg>
    </span>
    <input
      class="ui-checkbox__input"
      id="ui-checkbox-1"
      role="checkbox"
      type="checkbox"
      value=""
    />
  </div>
  <label
    class="ui-checkbox__label"
    for="ui-checkbox-1"
  >
    Checkbox label
  </label>
</div>
`;
